cmake/CMakeLists.txt

```diff
@@ -54,6 +54,7 @@ option(onnxruntime_USE_DNNL "Build with DNNL support" OFF)
 option(onnxruntime_USE_FEATURIZERS "Build ML Featurizers support" OFF)
 option(onnxruntime_DEV_MODE "Enable developer warnings and treat most of them as error." OFF)
 option(onnxruntime_BUILD_UNIT_TESTS "Build ONNXRuntime unit tests" ON)
+option(onnxruntime_DISABLE_UNIT_TESTS "Disable ONNXRuntime unit tests" OFF)
```

Comment:
Would it be too confusing if we have both onnxruntime_BUILD_UNIT_TESTS and onnxruntime_DISABLE_UNIT_TESTS ?

Reply:
I added `DISABLE_UNIT_TESTS` because some pipelines failed with `BUILD_UNIT_TESTS=OFF`. Let me try again. Thanks.